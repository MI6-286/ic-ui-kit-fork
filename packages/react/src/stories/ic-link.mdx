import {
  Meta,
  Stories,
  Markdown
} from "@storybook/blocks";
import readme from "../../../web-components/src/components/ic-link/readme.md";
import * as IcLinkStories from './ic-link.stories';

<Meta of={IcLinkStories} />

<Markdown>{readme}</Markdown>

<Stories />
