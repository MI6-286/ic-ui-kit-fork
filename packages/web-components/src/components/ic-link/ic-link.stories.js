import { html } from "lit-html";

const defaultArgs = {
  href: "/",
  target: "",
  monochrome: false,
  theme: "inherit",
};

export default {
  title: "Web Components/Link",
  component: "ic-link",
};

export const Default = {
  render: () =>
    html`<div>
      <ic-link href="/">This is a link</ic-link>
      <br />
      <br />
      <div style="width: 100px">
        <ic-link href="/"
          >This is a very very long link that goes multi-line</ic-link
        >
      </div>
    </div>`,

  name: "Default",

  parameters: {
    layout: "fullscreen",
  },
};

export const WithIcon = {
  render: () =>
    html`<ic-link href="/" target="_blank">This is a link</ic-link>`,
  name: "With icon",

  parameters: {
    layout: "fullscreen",
  },
};

export const DownloadLink = {
  render: () => html`<ic-link href="/" download>Download File</ic-link>`,
  name: "Download link",

  parameters: {
    layout: "fullscreen",
  },
};

export const Playground = {
  render: (args) =>
    html` <ic-link
      href=${args.href}
      target=${args.target}
      monochrome=${args.monochrome}
      theme=${args.theme}
      >This is a link</ic-link
    >`,

  args: defaultArgs,

  argTypes: {
    target: {
      options: ["_blank", "_self", "_parent", "_top", ""],

      control: {
        type: "radio",
      },
    },

    theme: {
      options: ["inherit", "light", "dark"],

      control: {
        type: "inline-radio",
      },
    },
  },

  name: "Playground",
};
