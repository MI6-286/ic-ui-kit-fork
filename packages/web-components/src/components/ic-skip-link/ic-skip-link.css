ic-link {
  position: absolute;
  left: -9999px;
}

ic-link:focus {
  z-index: var(--ic-z-index-skip-link);
  left: var(--ic-space-sm);
  padding: var(--ic-space-sm);
  box-shadow: var(--ic-border-focus);
  border-radius: var(--ic-border-radius);
}

.display-top:focus {
  top: var(--ic-space-sm);
}

.full-width:focus {
  display: flex;
  justify-content: center;
  right: var(--ic-space-sm);
}

.show-background {
  background: var(--ic-skip-link-background);
}
