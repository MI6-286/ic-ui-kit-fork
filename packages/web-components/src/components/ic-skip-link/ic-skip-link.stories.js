import { html } from "lit-html";

export default {
  title: "Web Components/Skip link",
  component: "ic-skip-link",
};

export const Default = {
  render: () =>
    html`<ic-skip-link target="#page-content"></ic-skip-link>
      <ic-top-navigation app-title="Application Name">
        <svg
          slot="app-icon"
          xmlns="http://www.w3.org/2000/svg"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
          fill="#000000"
        >
          <path d="M0 0h24v24H0V0z" fill="none" />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"
          />
        </svg>
      </ic-top-navigation>
      <main id="page-content">Target element</main>
      <ic-button>Should receive next focus</ic-button>`,
  name: "Default",
};

export const FullWidth = {
  render: () =>
    html`<ic-skip-link target="#page-content" full-width="true"></ic-skip-link>
      <ic-top-navigation app-title="Application Name">
        <svg
          slot="app-icon"
          xmlns="http://www.w3.org/2000/svg"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
          fill="#000000"
        >
          <path d="M0 0h24v24H0V0z" fill="none" />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"
          />
        </svg>
      </ic-top-navigation>
      <main id="page-content">Target element</main>
      <ic-button>Should receive next focus</ic-button>`,
  name: "Full width",
};

export const Inline = {
  render: () =>
    html`<ic-skip-link target="#page-content"></ic-skip-link>
      <ic-top-navigation app-title="Application Name">
        <svg
          slot="app-icon"
          xmlns="http://www.w3.org/2000/svg"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
          fill="#000000"
        >
          <path d="M0 0h24v24H0V0z" fill="none" />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"
          />
        </svg>
      </ic-top-navigation>
      <br />
      <br />
      <br />
      <main id="page-content">Target element</main>
      <ic-skip-link target="#next-content" inline="true"></ic-skip-link>
      <br />
      <br />
      <br />
      <main id="next-content">Next target element</main>
      <ic-button>Should receive last focus</ic-button>`,
  name: "Inline",
};

export const HideBackground = {
  render: () =>
    html`<ic-skip-link
        target="#page-content"
        transparent-background="true"
      ></ic-skip-link>
      <ic-top-navigation app-title="Application Name">
        <svg
          slot="app-icon"
          xmlns="http://www.w3.org/2000/svg"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
          fill="#000000"
        >
          <path d="M0 0h24v24H0V0z" fill="none" />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"
          />
        </svg>
      </ic-top-navigation>
      <main id="page-content">Target element</main>
      <ic-button>Should receive next focus</ic-button>`,
  name: "Hide background",
};

const defaultArgs = {
  fullWidth: false,
  inline: false,
  label: "Skip to main content",
  monochrome: false,
  target: "#",
  theme: "inherit",
  transparentBackground: false,
};

export const Playground = {
  render: (args) =>
    html`<ic-skip-link
        full-width=${args.fullWidth}
        inline=${args.inline}
        label=${args.label}
        monochrome=${args.monochrome}
        target=${args.target}
        theme=${args.theme}
        transparent-background=${args.transparentBackground}
      ></ic-skip-link>
      <ic-top-navigation app-title="Application Name">
        <svg
          slot="app-icon"
          xmlns="http://www.w3.org/2000/svg"
          height="24px"
          viewBox="0 0 24 24"
          width="24px"
          fill="#000000"
        >
          <path d="M0 0h24v24H0V0z" fill="none" />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm-5.5-2.5l7.51-3.49L17.5 6.5 9.99 9.99 6.5 17.5zm5.5-6.6c.61 0 1.1.49 1.1 1.1s-.49 1.1-1.1 1.1-1.1-.49-1.1-1.1.49-1.1 1.1-1.1z"
          />
        </svg>
      </ic-top-navigation>
      <main id=${args.target.replace("#", "")}>Target element</main>
      <ic-button>Should receive next focus</ic-button>`,

  args: defaultArgs,

  argTypes: {
    theme: {
      options: ["inherit", "light", "dark"],
      control: { type: "inline-radio" },
    },
  },

  name: "Playground",
};
