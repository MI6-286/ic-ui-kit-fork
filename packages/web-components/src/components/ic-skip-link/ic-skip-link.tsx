import { Component, Element, h, Host, Method, Prop } from "@stencil/core";
import { IcThemeMode } from "../../utils/types";

@Component({
  tag: "ic-skip-link",
  styleUrl: "ic-skip-link.css",
  shadow: true,
})
export class SkipLink {
  private linkEl?: HTMLIcLinkElement;

  @Element() el: HTMLIcSkipLinkElement;

  /**
   * If `true`, the skip link will fill the width of the page.
   */
  @Prop() fullWidth: boolean = false;

  /**
   * If `true`, the skip link will appear inline with surrounding page content when focused.
   */
  @Prop() inline: boolean = false;

  /**
   * The label displayed when the skip link is focused.
   */
  @Prop() label: string = "Skip to main content";

  /**
   * If `true`, the skip link will display as black in the light theme, and white in the dark theme.
   */
  @Prop() monochrome: boolean = false;

  /**
   * The target ID for the element which should receive focus when triggering the skip link.
   */
  @Prop() target!: string;

  /**
   * Sets the theme color to the dark or light theme color. `inherit` will set the color based on the system settings or ic-theme component.
   */
  @Prop() theme: IcThemeMode = "inherit";

  /**
   * If `true`, the background will be hidden.
   */
  @Prop() transparentBackground: boolean = false;

  /**
   * Sets focus on the element.
   */
  @Method()
  async setFocus(): Promise<void> {
    if (this.linkEl) this.linkEl.setFocus();
  }

  render() {
    const {
      fullWidth,
      inline,
      label,
      monochrome,
      target,
      theme,
      transparentBackground,
    } = this;

    return (
      <Host
        class={{
          [`ic-theme-${theme}`]: theme !== "inherit",
        }}
      >
        <ic-link
          class={{
            "display-top": !inline,
            "full-width": !!fullWidth,
            "show-background": !transparentBackground,
            inline: !!inline,
          }}
          href={`#${target}`}
          monochrome={monochrome}
          theme={theme}
          ref={(el) => (this.linkEl = el)}
        >
          {label}
        </ic-link>
      </Host>
    );
  }
}
