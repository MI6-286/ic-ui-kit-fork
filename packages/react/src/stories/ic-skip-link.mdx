import { Markdown, Meta, Stories } from "@storybook/blocks";
import * as IcSkipLinkStories from './ic-skip-link.stories';
import readme from "../../../web-components/src/components/ic-skip-link/readme.md";

<Meta of={IcSkipLinkStories} />

<Markdown>{readme}</Markdown>

<Stories />
