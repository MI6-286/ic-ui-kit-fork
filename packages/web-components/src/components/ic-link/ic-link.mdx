import { Meta, Stories, Markdown } from "@storybook/blocks";
import readme from "./readme.md";
import * as IcLinkStories from "./ic-link.stories";

<Meta of={IcLinkStories} />

<Markdown>{readme}</Markdown>

<Stories />
