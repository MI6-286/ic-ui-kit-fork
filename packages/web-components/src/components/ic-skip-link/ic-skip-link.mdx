import { Markdown, Meta, Stories } from "@storybook/blocks";
import * as IcSkipLinkStories from "./ic-skip-link.stories";
import readme from "./readme.md";

<Meta of={IcSkipLinkStories} />

<Markdown>{readme}</Markdown>

<Stories />
