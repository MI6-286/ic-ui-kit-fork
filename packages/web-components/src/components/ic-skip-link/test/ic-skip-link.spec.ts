import { newSpecPage } from "@stencil/core/testing";
import { Link } from "../../ic-link/ic-link";
import { SkipLink } from "../ic-skip-link";

describe("ic-skip-link component", () => {
  it("should render a skip link by default with the correct text", async () => {
    const page = await newSpecPage({
      components: [SkipLink, Link],
      html: `<ic-skip-link target="page-content"></ic-skip-link>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render a skip link with a custom label", async () => {
    const page = await newSpecPage({
      components: [SkipLink, Link],
      html: `<ic-skip-link target="page-content" label="Custom skip label"></ic-skip-link>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render a skip link without a background when `transparentBackground` is true", async () => {
    const page = await newSpecPage({
      components: [SkipLink, Link],
      html: `<ic-skip-link target="page-content" transparent-background="true"></ic-skip-link>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should pass down the theme and monochrome prop to the ic-link component", async () => {
    const page = await newSpecPage({
      components: [SkipLink, Link],
      html: `<ic-skip-link target="page-content" theme="dark" monochrome="true"></ic-skip-link>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with the `inline` prop", async () => {
    const page = await newSpecPage({
      components: [SkipLink, Link],
      html: `<ic-skip-link target="page-content" inline="true"></ic-skip-link>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
