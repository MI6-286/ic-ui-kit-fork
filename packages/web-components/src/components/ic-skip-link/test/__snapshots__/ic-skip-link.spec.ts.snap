// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-skip-link component should pass down the theme and monochrome prop to the ic-link component 1`] = `
<ic-skip-link class="ic-theme-dark" monochrome="true" target="#page-content" theme="dark">
  <mock:shadow-root>
    <ic-link class="display-top ic-link ic-link-monochrome ic-theme-dark show-background" exportparts="link">
      <mock:shadow-root>
        <a class="link" href="#page-content" part="link" tabindex="0">
          <slot></slot>
        </a>
      </mock:shadow-root>
      Skip to main content
    </ic-link>
  </mock:shadow-root>
</ic-skip-link>
`;

exports[`ic-skip-link component should render a skip link by default with the correct text 1`] = `
<ic-skip-link target="#page-content">
  <mock:shadow-root>
    <ic-link class="display-top ic-link show-background" exportparts="link">
      <mock:shadow-root>
        <a class="link" href="#page-content" part="link" tabindex="0">
          <slot></slot>
        </a>
      </mock:shadow-root>
      Skip to main content
    </ic-link>
  </mock:shadow-root>
</ic-skip-link>
`;

exports[`ic-skip-link component should render a skip link with a custom label 1`] = `
<ic-skip-link label="Custom skip label" target="#page-content">
  <mock:shadow-root>
    <ic-link class="display-top ic-link show-background" exportparts="link">
      <mock:shadow-root>
        <a class="link" href="#page-content" part="link" tabindex="0">
          <slot></slot>
        </a>
      </mock:shadow-root>
      Custom skip label
    </ic-link>
  </mock:shadow-root>
</ic-skip-link>
`;

exports[`ic-skip-link component should render a skip link without a background when \`transparentBackground\` is true 1`] = `
<ic-skip-link target="#page-content" transparent-background="true">
  <mock:shadow-root>
    <ic-link class="display-top ic-link" exportparts="link">
      <mock:shadow-root>
        <a class="link" href="#page-content" part="link" tabindex="0">
          <slot></slot>
        </a>
      </mock:shadow-root>
      Skip to main content
    </ic-link>
  </mock:shadow-root>
</ic-skip-link>
`;

exports[`ic-skip-link component should render with the \`inline\` prop 1`] = `
<ic-skip-link inline="true" target="#page-content">
  <mock:shadow-root>
    <ic-link class="ic-link show-background" exportparts="link">
      <mock:shadow-root>
        <a class="link" href="#page-content" part="link" tabindex="0">
          <slot></slot>
        </a>
      </mock:shadow-root>
      Skip to main content
    </ic-link>
  </mock:shadow-root>
</ic-skip-link>
`;
